diff --git a/src/AsyncSocket.h b/src/AsyncSocket.h
index 743f406..f8d1c6b 100644
--- a/src/AsyncSocket.h
+++ b/src/AsyncSocket.h
@@ -24,6 +24,8 @@
  * to unsigned length for everything to/from uSockets - this would however remove the opportunity
  * to signal error with -1 (which is how the entire UNIX syscalling is built). */
 
+#include "libusockets.h"
+
 #include "LoopData.h"
 #include "AsyncSocketData.h"
 
diff --git a/src/WebSocketContextData.h b/src/WebSocketContextData.h
index 269bf70..68dc956 100644
--- a/src/WebSocketContextData.h
+++ b/src/WebSocketContextData.h
@@ -24,6 +24,7 @@
 #include "WebSocketProtocol.h"
 #include "TopicTree.h"
 #include "WebSocketData.h"
+#include "Loop.h"
 
 namespace uWS {
 
diff --git a/src/WebSocketProtocol.h b/src/WebSocketProtocol.h
index db282e4..2e4ddd5 100644
--- a/src/WebSocketProtocol.h
+++ b/src/WebSocketProtocol.h
@@ -259,7 +259,7 @@ static inline size_t formatMessage(char *dst, const char *src, size_t length, Op
 
 // essentially this is only a parser
 template <const bool isServer, typename Impl>
-struct WIN32_EXPORT WebSocketProtocol {
+struct WebSocketProtocol {
 public:
     static const unsigned int SHORT_MESSAGE_HEADER = isServer ? 6 : 2;
     static const unsigned int MEDIUM_MESSAGE_HEADER = isServer ? 8 : 4;
